### PR TITLE
Add disclosure input to Kirby item 

### LINKED ISSUE #2560: [Enhancement] Item disclosure exploration
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
We want more affordance on our list items, to make sure that users know that some items i clickable.

## Describe the solution you'd like
Experiment with adding an ikon in the end of items (arrow right, arrow down, out of app )
One solution could be a new end disclosure slot.
Discuss more details with UX

## Are there any additional context?
Slice (8)

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by AgreSanGit**
## Which issue does this PR close?

This PR closes #2560 

## What is the new behavior?

Kirby item now supports a new input "disclosure" which is a special input used for setting navigation icons. This PR also documents this new feature in Kirby cookbook. 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/designsystem/src/lib/components/item/item.component.ts [hunk 1]
```diff
@@ -25,6 +25,9 @@ export class ItemComponent {
   @HostBinding('class.selected')
   @Input()
   selected: boolean;
+  @HostBinding('class.disclosure')
+  @Input()
+  disclosure: 'link' | 'arrow-more' | 'arrow-down' | 'arrow-up';
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/item/item.component.ts:22]: We should discuss with UX if we want to always set the 'selected' input when disclosure is true. 

If not, we should add to the documentation that it is best practice to also set selectable.

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 2]
```diff
@@ -15,33 +15,112 @@
 
 <div class="item-illustration">
   <svg
-    width="100%"
-    height="100%"
-    viewBox="0 0 343 56"
-    fill="none"
     xmlns="http://www.w3.org/2000/svg"
+    xmlns:xlink="http://www.w3.org/1999/xlink"
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: even though the new svg is beautiful, we need to revert now that disclosure is no longer a slot 🥲

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 3]
```diff
@@ -67,6 +146,10 @@
   The <em>outside</em> slot can be used for flags as in
   <a href="#" (click)="scrollTo(flagged)">this example</a>.
 </p>
+<p>
+  The <em>disclosure</em> slot can be used for navigation icons which are toned down to Kirby grey-4
+  as in <a href="#" (click)="scrollTo(disclosure)">this example</a>.
+</p>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]:
```suggestion

```

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 4]
```diff
@@ -211,6 +294,18 @@ <h3>Numeric</h3>
...
+<cookbook-example-viewer [html]="disclosureExample.template">
+  <cookbook-item-example-disclosure #disclosureExample></cookbook-item-example-disclosure>
+</cookbook-example-viewer>
+<p class="comment small">
+  <em
+    ><strong>Please note</strong>, set disclosure attribute on <code>kirby-item</code> to current
+    available icons in disclosure slot
+    <code> 'link' | 'arrow-more' | 'arrow-down' | 'arrow-up'</code>
+  </em>
+</p>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]:
```suggestion
    ><strong>Please note</strong>, the disclosure input is limited to a set of predefined icons: <code> 'link' | 'arrow-more' | 'arrow-down' | 'arrow-up'</code>
  </em>
</p>
```

Consider something like this instead.

File: libs/designsystem/src/lib/components/item/item.component.scss [hunk 5]
```diff
@@ -88,6 +88,14 @@
   }
 }
 
+.disclosure {
+  text-align: center;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/item/item.component.scss]: why are we setting text-align here? To me it seems like it has no effect, it is overwritten by ionic-specific text-align anyway.

image

File: libs/designsystem/src/lib/components/item/item.component.scss [hunk 6]
```diff
@@ -88,6 +88,14 @@
...
+  height: 24px;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/item/item.component.scss]: Why hardcode a height here? I suspect this is not necessary.

File: libs/designsystem/src/lib/components/item/item.component.scss [hunk 7]
```diff
@@ -107,6 +115,12 @@
   }
 }
 
+:host(.disclosure) {
+  ion-item {
+    --inner-padding-end: 0px;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/item/item.component.scss]: Could we not just set this to `#{utils.size('xxs')}` and avoid setting padding-right on the icon?

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Really really great work, I bring only minor comments 🐼 

Couple of general thoughts:
1. Now would be a great time to add an API description table to the item showcase, and add the disclosure input there too
2. Would be good to add a test or two that verifying correct paddings when disclosure is set.

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html (hunk 2 above)
Comment (AgreSanGit) [reply apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Reverted 🥲

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html (hunk 3 above)
Comment (AgreSanGit) [reply apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: No more

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html (hunk 4 above)
Comment (AgreSanGit) [reply apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Suggestion taken!

File: libs/designsystem/src/lib/components/item/item.component.scss (hunk 5 above)
Comment (AgreSanGit) [reply libs/designsystem/src/lib/components/item/item.component.scss]: Removed

File: libs/designsystem/src/lib/components/item/item.component.scss (hunk 6 above)
Comment (AgreSanGit) [reply libs/designsystem/src/lib/components/item/item.component.scss]: Removed

File: libs/designsystem/src/lib/components/item/item.component.scss (hunk 7 above)
Comment (AgreSanGit) [reply libs/designsystem/src/lib/components/item/item.component.scss]: You're right! fixed

**Review by RasmusKjeldgaard — DISMISSED** [review]

**Comment by RasmusKjeldgaard** [comment]
Remember to change the PR description so it no longer says slot.

File: libs/designsystem/src/lib/components/item/item.component.ts (hunk 1 above)
Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/item/item.component.ts:22]: selectable, not selected

File: apps/cookbook/src/app/showcase/page-showcase/page-showcase.component.html [hunk 8]
```diff
@@ -56,11 +56,15 @@ <h3>Life-cycle hooks</h3>
   <p>
     The <code>kirby-router-outlet</code> stacks the pages/components when it navigates. This means
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/page-showcase/page-showcase.component.html]: Revert the changes to this file as they are not related to the issue at hand 😃

File: libs/designsystem/src/lib/components/item/item.component.spec.ts [hunk 9]
```diff
@@ -82,4 +82,15 @@ describe('ItemComponent', () => {
       expect(spectator.query('ion-item')).toHaveComputedStyle({ '--padding-bottom': size('xxs') });
     });
   });
+
+  describe('when disclosure is set to something valid', () => {
+    it('should Inner-padding-end be 8px', () => {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/item/item.component.spec.ts]:
```suggestion
    it('should set inner-padding-end to xxs', () => {
```

**Review by RasmusKjeldgaard — CHANGES_REQUESTED** [review]
Just a couple of small edits needed before merge

File: apps/cookbook/src/app/showcase/page-showcase/page-showcase.component.html (hunk 8 above)
Comment (AgreSanGit) [reply apps/cookbook/src/app/showcase/page-showcase/page-showcase.component.html]: Yes sir should be fixed

**Review by RasmusKjeldgaard — APPROVED** [review]

**Comment by RasmusKjeldgaard** [comment]
Looks really good, great job!
